influxdb3_write/src/write_buffer/persisted_files.rs

```diff
@@ -93,4 +54,229 @@ impl PersistedFiles {
+    }
+
+    #[test_log::test(test)]
+    fn test_get_metrics_after_update_with_duplicate_file() {
+        let all_persisted_snapshot_files = build_persisted_snapshots();
+        let already_existing_file = all_persisted_snapshot_files
+            .last()
+            .unwrap()
+            .databases
+            .get("db-1")
+            .unwrap()
+            .tables
+            .get("table-1")
+            .unwrap()
+            .last()
+            .cloned()
+            .unwrap();
+
+        let persisted_file =
+            PersistedFiles::new_from_persisted_snapshots(all_persisted_snapshot_files);
+        let mut parquet_files = build_parquet_files(4);
+        info!(all_persisted_files = ?persisted_file, "Full persisted file");
+        info!(already_existing_file = ?already_existing_file, "Existing file");
+        parquet_files.push(already_existing_file);
+
+        let new_snapshot = build_snapshot(parquet_files, 1, 1, 1);
+        persisted_file.add_persisted_snapshot_files(new_snapshot);
+
+        let (file_count, size_in_mb, row_count) = persisted_file.get_metrics();
+        info!(all_persisted_files = ?persisted_file, "Full persisted file after");
+
+        info!(metrics = ?persisted_file.get_metrics(), "All files metrics");
+        assert_eq!(14, file_count);
+        // TODO: Just tying in TODO within build_snapshot function below. Even though
+        //       there are only 14 files added to persisted_file the below 2 metrics
+        //       are for 15 files because of using `add_parquet_file` directly which
+        //       doesn't check for duplicates
+        assert_eq!(0.75, size_in_mb);
+        assert_eq!(150, row_count);
+    }
+
+    fn build_persisted_snapshots() -> Vec<PersistedSnapshot> {
+        let mut all_persisted_snapshot_files = Vec::new();
+        let parquet_files_1 = build_parquet_files(5);
+        all_persisted_snapshot_files.push(build_snapshot(parquet_files_1, 1, 1, 1));
+
+        let parquet_files_2 = build_parquet_files(5);
+        all_persisted_snapshot_files.push(build_snapshot(parquet_files_2, 2, 2, 2));
+
+        all_persisted_snapshot_files
+    }
+
+    fn build_snapshot(
+        parquet_files: Vec<ParquetFile>,
+        snapshot_id: u64,
+        wal_id: u64,
+        catalog_id: u32,
+    ) -> PersistedSnapshot {
+        let snap1 = SnapshotSequenceNumber::new(snapshot_id);
+        let wal1 = WalFileSequenceNumber::new(wal_id);
+        let cat1 = SequenceNumber::new(catalog_id);
+        let mut new_snapshot =
+            PersistedSnapshot::new("dummy-host-id".to_owned(), snap1, wal1, cat1);
+        parquet_files.into_iter().for_each(|file| {
+            // TODO: Check why `add_parquet_file` method does not check if file is
+            //       already present. This is checked when trying to add a new PersistedSnapshot
+            //       as part of snapshotting process.
+            new_snapshot.add_parquet_file(Arc::from("db-1"), Arc::from("table-1"), file);
```

Comment:
When adding snapshot to `PersistedFile` ([code](https://github.com/influxdata/influxdb/blob/7d37bbbce7ecd4f6c95445de2b22ee189dd93b85/influxdb3_write/src/write_buffer/persisted_files.rs#L74)) we seem to check for the duplicate but this method does not do any checks and adds it directly. It's only for the test setup, but wanted to confirm if using this method (`add_parquet_fie`) here is fine or not.